File: backend/models/Workout.js
const mongoose = require('mongoose');

const workoutSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  type: { type: String, required: true },
  duration: { type: Number, required: true },
  calories: { type: Number },
  distance: { type: Number },
  date: { type: Date, default: Date.now },
});

module.exports = mongoose.model('Workout', workoutSchema);

File: frontend/lib/models/workout.dart
class Workout {
  final String? id;
  final String type;
  final int duration;
  final int? calories;
  final double? distance;
  final DateTime date;

  Workout({
    this.id,
    required this.type,
    required this.duration,
    this.calories,
    this.distance,
    required this.date,
  });

  factory Workout.fromJson(Map<String, dynamic> json) {
    return Workout(
      id: json['_id'],
      type: json['type'],
      duration: json['duration'],
      calories: json['calories'],
      distance: json['distance']?.toDouble(),
      date: DateTime.parse(json['date']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) '_id': id,
      'type': type,
      'duration': duration,
      if (calories != null) 'calories': calories,
      if (distance != null) 'distance': distance,
      'date': date.toIso8601String(),
    };
  }
}